secant/Utils/Zatoshi.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Zatoshi.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 26.05.2022.
+//
+
+import Foundation
+
+typealias ZEC = NSDecimalNumber
+
+// MARK: - Zatoshi
+
+struct Zatoshi {
+    @Clamped(-(100_000_000 * 21_000_000)...(100_000_000 * 21_000_000))
+    var amount: Int64 = 0
+
+    /// Converts `Zatoshi` to `ZEC`
+    var asZec: ZEC {
+        ZEC(value: Double(amount) / Double(100_000_000))
+    }
+    
+    /// The value of `Zatoshi` is valid when zero or positive number. User can't send or receive negative amount of `Zatoshi`.
+    /// Negative `Zatoshi` makes sense only as some partial result, generally speaking during some computations.
+    /// Another example would be summing up last 10 transactions resulting in negative `Zatoshi` value.
+    var isValid: Bool {
+        amount >= 0
+    }
+    
+    static func + (left: Zatoshi, right: Zatoshi) -> Zatoshi {
+        Zatoshi(amount: left.amount + right.amount)
+    }
+
+    static func - (left: Zatoshi, right: Zatoshi) -> Zatoshi {
+        Zatoshi(amount: left.amount - right.amount)
+    }
+}
+
+extension Int {
+    var asZatoshi: Zatoshi { Zatoshi(amount: Int64(self)) }
+    var asZec: ZEC { ZEC(value: Double(self)) }
+}
+
+extension Int64 {
+    var asZatoshi: Zatoshi { Zatoshi(amount: self) }
+    var asZec: ZEC { ZEC(value: Double(self)) }
+}
+
+// MARK: - ZEC
+
+extension ZEC {
+    static var nsDecimalHandler = NSDecimalNumberHandler(
+        roundingMode: ZEC.RoundingMode.plain,
+        scale: 8,
+        raiseOnExactness: true,
+        raiseOnOverflow: true,
+        raiseOnUnderflow: true,
+        raiseOnDivideByZero: true
+    )
+    
+    var humanReadable: ZEC {
+        self.rounding(accordingToBehavior: ZEC.nsDecimalHandler)
+    }
+    
+    /// Converts `ZEC` to `Zatoshi`
+    var asZatoshi: Zatoshi {
+        Zatoshi(amount: (self * NSDecimalNumber(value: 100_000_000)).int64Value)
+    }
+
+    static func + (left: ZEC, right: ZEC) -> ZEC {
```

Comment:
I don't see where we could use these math operators.  Am I missing something?

Reply:
removed